### DISCUSSION TITLE
nerdctl failed to run images by error: FATA[0005] mount callback failed on /run/user/0/containerd-mount3572667506: read /run/user/0/containerd-mount3572667506/etc/group: input/output error

### DISCUSSION THREAD

**Discussion by Curtis2Origin** [containerd/nerdctl#2343] [Q&A]
Hi everyone, 

I conduct the experiment according to the following article(https://d7y.io/zh/docs/setup/integration/nydus). And an error occurred when I reached this step: 
nerdctl --snapshotter nydus run --rm -it $DOCKERHUB_REPO_NAME/python:3.9.15-nydus
Error msg:
FATA[0005] mount callback failed on /run/user/0/containerd-mount3572667506: read /run/user/0/containerd-mount3572667506/etc/group: input/output error

Does anyone know how to solve this problem?

[ENV]
os: centos 7.3, kernel-4.18
nerdctl version 1.4.0
containerd version: Version:  v1.7.1
docker version: 18.06.3-ce
nydus-snapshotter-v0.3.3-x86_64.tgz
nydus-static-v2.1.1-linux-amd64.tgz

**Comment by AkihiroSuda** [comment]
Is this specific to Nydus?

**Reply by Curtis2Origin** [reply]
Hi AkihiroSuda,

Is it a containerd's defect? From containerd source code, it's seemed that failed at func WithTempMount. And the default tempDir is set via env XDG_RUNTIME_DIR(default is /run/user/0). Is there any work around to skip that issue?

By the way, I checked this issue https://github.com/containerd/nerdctl/issues/690. Is it the same issue?

Thanks
